### PR TITLE
Cleanvision Integration

### PR DESCRIPTION
**Pull request by sanjanag**
Integrated cleanvision in Datalab
- API change - `Datalab(data, label_name, image_key)` where image_key is used to instantiate imagelab
- To check for specific imagelab issue types, user can supply `issue_types = {"image_issue_types": {"dark": {}}}`, image_issue_types key encapsulates all arguments of imagelab
- Created an adapter module `cleanlab/datalab/adapter/imagelab.py` that adapts important classes `ImagelabDataIssuesAdapter`, `ImagelabReporterAdapter`, `ImagelabIssueFinderAdapter` to work for both imagelab and datalab
- `cleanlab/datalab/helper_factory.py` has factory methods that pick the appropriate classes if imagelab is present

- [x] Update ci.yml and setup.py after cleanvision release with a specific version

https://github.com/sanjanag/test_datalab/blob/main/worfklows.ipynb

### CODE REVIEW

File: cleanlab/datalab/issue_finder.py [hunk 1]
```diff
@@ -69,8 +70,9 @@ class IssueFinder:
     `Datalab.find_issues` method which internally utilizes an IssueFinder instance.
     """
 
-    def __init__(self, datalab: "Datalab", verbosity=1):
+    def __init__(self, datalab: "Datalab", imagelab: "Imagelab", verbosity=1):
         self.datalab = datalab
+        self.imagelab = imagelab
```
Comment (elisno) [inline cleanlab/datalab/issue_finder.py:75]: No need to pass `imagelab` as a parameter to the constructor. It's already a field of `datalab`.

File: cleanlab/datalab/issue_finder.py [hunk 2]
```diff
@@ -157,12 +153,16 @@ def find_issues(
...
-            for factory in _IssueManagerFactory.from_list(list(issue_types_copy.keys()))
-        ]
+        new_issue_managers = []
+        for issue_type in issue_types_copy.keys():
+            if issue_type == "image_issue_types":
+                continue
+            factory = _IssueManagerFactory.from_str(issue_type)
+            new_issue_managers.append(
+                factory(datalab=self.datalab, **issue_types_copy.get(factory.issue_name, {}))
+            )
```
Comment (elisno) [inline cleanlab/datalab/issue_finder.py:163]: I think we can revert this block, but make sure that `issue_types_copy` won't break the factory for the Datalab-specific issue managers.

File: cleanlab/datalab/issue_finder.py [hunk 3]
```diff
@@ -182,12 +182,29 @@ def find_issues(
...
+                print(
+                    f'Finding {", ".join(issue_types_copy["image_issue_types"].keys())} images ...'
+                )
+            self.imagelab.find_issues(issue_types=issue_types_copy["image_issue_types"])
+            data_issues.collect_statistics_from_issue_manager(self.imagelab)
+            data_issues._collect_results_from_imagelab(self.imagelab)
+        except Exception as e:
+            print(f"Error in checking for image issues: {e}")
+            failed_managers.append(self.imagelab)
+
```
Comment (elisno) [inline cleanlab/datalab/issue_finder.py:197]: This try-block is nearly identical to the one in the for-loop above, except:

- `issue_manager` becomes `self.imagelab`.

If we make adapter classes to these methods (in `DataIssues`). Then we can just make an adpter class for IssueFinder as well, that only needs to define `find_issues`.

**Review by elisno — COMMENTED** [review]
Some comments on a potential refactor.

File: cleanlab/datalab/adapter/imagelab.py [hunk 4]
```diff
@@ -0,0 +1,196 @@
+"""Module for adapting Imagelab to the Datalab API.
+
+It uses the Adapter pattern so that Imagelab becomes compatible
+with the DataIssues, IssueFinder, Reporter classes.
```
Comment (jwmueller) [inline cleanlab/datalab/adapter/imagelab.py:4]: Is this something users would ever care about?

For developer notes, I would explicitly write: 
**Note for developers:** ... 

and assume the documentation is otherwise intended for users.

File: cleanlab/datalab/adapter/imagelab.py [hunk 5]
```diff
@@ -0,0 +1,196 @@
+"""Module for adapting Imagelab to the Datalab API.
```
Comment (jwmueller) [inline cleanlab/datalab/adapter/imagelab.py:1]:
```suggestion
"""A wrapper around the Imagelab class from the CleanVision package to incorporate it into Datalab. This allows low-quality images to be detected alongside other issues in computer vision datasets.
```

File: cleanlab/datalab/adapter/imagelab.py [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+    """
+    imagelab = None
+    if not image_key:
+        return imagelab
+    try:
+        from cleanvision.imagelab import Imagelab
+        from datasets.arrow_dataset import Dataset
+
+        if isinstance(dataset, Dataset):
+            imagelab = Imagelab(hf_dataset=dataset, image_key=image_key, verbosity=0)
```
Comment (jwmueller) [inline cleanlab/datalab/adapter/imagelab.py:50]: what if the key does not exist in the dataset? will cleanvision output a helpful error message?

File: cleanlab/datalab/adapter/imagelab.py [hunk 7]
```diff
@@ -0,0 +1,196 @@
...
+        else:
+            raise ValueError(
+                "Only huggingface datasets are supported for cleanvision checks from cleanlab as of now"
```
Comment (jwmueller) [inline cleanlab/datalab/adapter/imagelab.py:53]:
```suggestion
                "For now, only huggingface datasets are supported for running cleanvision checks inside cleanlab. You can easily convert most datasets to the huggingface dataset format."
```

File: cleanlab/datalab/adapter/imagelab.py [hunk 8]
```diff
@@ -0,0 +1,196 @@
...
+            )
+
+    except ImportError:
+        raise ImportError(
+            "Cannot import datasets or cleanvision package. Please install them and try again, or just install cleanlab with "
```
Comment (jwmueller) [inline cleanlab/datalab/adapter/imagelab.py:58]:
```suggestion
            "Cannot import required packages: datasets, cleanvision. Please install them and try again, or just install cleanlab with "
```

File: cleanlab/datalab/adapter/imagelab.py (hunk 6 above)
Comment (jwmueller) [reply cleanlab/datalab/adapter/imagelab.py:50]: make sure in this edge-case that the user does not get the "Cannot import datasets or cleanvision package..." error message

File: tests/datalab/test_datalab.py [hunk 9]
```diff
@@ -19,20 +19,19 @@
...
 import numpy as np
 import pandas as pd
-
-from pathlib import Path
-
 import pytest
-import timeit
+from datasets.dataset_dict import DatasetDict
+from scipy.sparse import csr_matrix
+from sklearn.neighbors import NearestNeighbors
```
Comment (jwmueller) [inline tests/datalab/test_datalab.py:31]: There needs to be a new unit test for the cleanvision integration. 
It needs to create a toy image dataset and verify cleanvision was actually able detect at least one type of issue (that you included in this toy image dataset).

File: cleanlab/datalab/data_issues.py [hunk 10]
```diff
@@ -217,10 +220,55 @@ def collect_statistics_from_issue_manager(self, issue_manager: IssueManager) ->
 
         """
         key = "statistics"
-        statistics: Dict[str, Any] = issue_manager.info.pop(key, {})
+        # todo: check if the code breaks between get and pop
```
[the remark this answers is not in the document]
Comment (sanjanag) [reply cleanlab/datalab/data_issues.py:211]: It doesn't affect as new issue managers are created on find_issues call

File: tests/datalab/test_datalab.py [hunk 11]
```diff
@@ -676,7 +689,7 @@ def test_report_for_outlier_issues_via_pred_probs(find_issues_kwargs):
     find_issues_kwargs["issue_types"] = {"outlier": {"k": 1}}
     lab.find_issues(**find_issues_kwargs)
 
-    reporter = Reporter(lab.data_issues, verbosity=0, include_description=False)
+    reporter = Reporter(lab.data_issues, imagelab=None, verbosity=0, include_description=False)
```
Comment (elisno) [inline tests/datalab/test_datalab.py:692]: If we decouple `Reporter` from `Imagelab`, then maybe this argument won't be necessary.

File: tests/datalab/test_report.py [hunk 12]
```diff
@@ -26,15 +26,16 @@ def data_issues(self, lab):
 
     @pytest.fixture
     def reporter(self, data_issues):
-        return Reporter(data_issues=data_issues)
+        return Reporter(data_issues=data_issues, imagelab=None)
```
Comment (elisno) [inline tests/datalab/test_report.py:29]: Previous comment applies here as well.

**Review by elisno — COMMENTED** [review]
Just some comments on the source code. Looking at the notebook next.

File: cleanlab/datalab/datalab.py [hunk 13]
```diff
@@ -308,11 +329,18 @@ def report(
...
+                "Please specify some issue types to run in datalab.find_issues() to see the report.\n"
+            )
+            return
+
+        reporter = report_factory(self._imagelab)(
             data_issues=self.data_issues,
             verbosity=verbosity,
             include_description=include_description,
             show_summary_score=show_summary_score,
+            imagelab=self._imagelab,
```
[the remark this answers is not in the document]
Comment (elisno) [reply cleanlab/datalab/datalab.py:343]: If it's not easy, then you can ignore the rest of my comments.

File: cleanlab/datalab/report.py [hunk 14]
```diff
@@ -101,6 +104,15 @@ def get_report(self, num_examples: int) -> str:
...
 
+        if self.imagelab:
+            issue_types = [
+                issue_type
+                for issue_type in issue_summary_sorted["issue_type"].tolist()
+                if issue_type not in self.imagelab.issue_summary["issue_type"].tolist()
+            ]
+        else:
+            issue_types = issue_summary_sorted["issue_type"].tolist()
+
```
[the remark this answers is not in the document]
Comment (sanjanag) [reply cleanlab/datalab/report.py:115]: good idea

File: cleanlab/datalab/datalab.py (hunk 13 above)
[the remark this answers is not in the document]
Comment (sanjanag) [reply cleanlab/datalab/datalab.py:343]: This is not possible, because factory would only return the reporter class, it would still need imagelab object to print its report. But I have removed it as from the `Reporter` class

File: cleanlab/datalab/adapter/imagelab.py (hunk 6 above)
Comment (sanjanag) [reply cleanlab/datalab/adapter/imagelab.py]: Checked this edge case. It is handled in cleanvision where Imagelab throws error if key is not found and this error is propagated in datalab.

**Review by elisno — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 15]
```diff
@@ -26,6 +26,10 @@ jobs:
         if: matrix.os != 'windows-latest'
         run: |
           pip install fasttext
+      - name: Install cleanvision
+        run: |
+          git clone https://github.com/cleanlab/cleanvision.git
+          pip install cleanvision/
```
Comment (jwmueller) [inline .github/workflows/ci.yml:32]: TODO: don't forget to update prior to merge

File: .github/workflows/ci.yml [hunk 16]
```diff
@@ -122,8 +134,12 @@ jobs:
...
-          pip install .  # install dependencies
+          pip install ".[all]"  # install dependencies
           pip install -r requirements-dev.txt  # install development dependencies and type stubs
+      - name: Install cleanvision
+        run: |
+          git clone https://github.com/cleanlab/cleanvision.git
+          pip install cleanvision/
       - name: Type check
         run: mypy --install-types --non-interactive cleanlab
   fmt:
```
Comment (jwmueller) [inline .github/workflows/ci.yml:145]: can you confirm none of your changes will affect the CI test we had before that checks cleanlab works without optional dependencies installed? thanks!

File: cleanlab/datalab/data_issues.py [hunk 17]
```diff
@@ -207,11 +208,27 @@ def collect_statistics_from_issue_manager(self, issue_manager: IssueManager) ->
...
```
Comment (jwmueller) [inline cleanlab/datalab/data_issues.py:211]: why is this Todo still here?

File: cleanlab/datalab/data_issues.py [hunk 18]
```diff
@@ -22,20 +22,21 @@
 instances and keeps track of each issue, a summary for each type of issue,
 related information and statistics about the issues.
 
-The collected information can be accessed using the 
+The collected information can be accessed using the
 :py:meth:`get_info <cleanlab.datalab.data_issues.DataIssues.get_info>` method.
```
Comment (jwmueller) [inline cleanlab/datalab/data_issues.py:26]:
```suggestion
The collected information can be accessed using the
:py:meth:`get_info <cleanlab.datalab.data_issues.DataIssues.get_info>` method.
We recommend using that method instead of this module, which is just intended for internal use.
```

File: cleanlab/datalab/datalab.py [hunk 19]
```diff
@@ -97,10 +105,12 @@ def __init__(
         self._label_map = self._labels.label_map
         self.label_name = self._labels.label_name
         self._data_hash = self._data._data_hash
-        self.data_issues = DataIssues(self._data)
         self.cleanlab_version = cleanlab.version.__version__
         self.verbosity = verbosity
+        self._imagelab = create_imagelab(dataset=self.data, image_key=image_key)
+        self.data_issues = data_issues_factory(self._imagelab)(self._data)
 
+    # todo: check displayer methods
```
Comment (jwmueller) [inline cleanlab/datalab/datalab.py:113]: why is this todo still here?

File: cleanlab/datalab/datalab.py [hunk 20]
```diff
@@ -270,13 +280,24 @@ def find_issues(
                 >>> # lab.find_issues(pred_probs=pred_probs, issue_types=issue_types)
 
         """
-        issue_finder = IssueFinder(datalab=self, verbosity=self.verbosity)
+        if issue_types is not None and not issue_types:
+            warnings.warn("No issue types were specified. No issues will be found in the dataset.")
```
Comment (jwmueller) [inline cleanlab/datalab/datalab.py:284]:
```suggestion
            warnings.warn("No issue types were specified so no issues will be found in the dataset. Set `issue_types` as None to consider a default set of issues.")
```

File: cleanlab/datalab/datalab.py [hunk 21]
```diff
@@ -270,13 +280,24 @@ def find_issues(
...
+            return None
+        no_args_passed = all(arg is None for arg in [pred_probs, features, knn_graph, issue_types])
+        if no_args_passed:
+            warnings.warn("No arguments were passed to Datalab.find_issues().")
```
Comment (jwmueller) [inline cleanlab/datalab/datalab.py:288]: CC @elisno 

Message should clarify: 

1. what is the implication of this 
2. what should user consider doing instead

File: cleanlab/datalab/datalab.py [hunk 22]
```diff
@@ -308,11 +329,18 @@ def report(
         """
         if verbosity is None:
             verbosity = self.verbosity
-        reporter = Reporter(
+        if self.data_issues.issue_summary.empty:
+            print(
+                "Please specify some issue types to run in datalab.find_issues() to see the report.\n"
```
Comment (jwmueller) [inline cleanlab/datalab/datalab.py:334]:
```suggestion
                "Please specify some `issue_types` in datalab.find_issues() to see a report.\n"
```

File: tests/datalab/test_cleanvision_integration.py [hunk 23]
```diff
@@ -0,0 +1,165 @@
+import numpy as np
```
Comment (jwmueller) [inline tests/datalab/test_cleanvision_integration.py:1]: where in this file is there any unit test of correct functionality? Need to have at least one unit test that serves as end-to-end check that functionality actually works as desired. Eg. have a toy dataset with 4 images, one dark, and show datalab flags only this one image with the dark issue

File: setup.py [hunk 24]
```diff
@@ -34,8 +34,15 @@ def run(self):
     "datasets>=2.7.0",
 ]
 
+CLEANVISION_REQUIRE = [
+    # Mainly for Datalab's data storage class.
+    # Still some type hints that require datasets
+    "cleanvision==0.3.0",
```
Comment (jwmueller) [inline setup.py:40]: this should be lower bound right?

File: setup.py [hunk 25]
```diff
@@ -34,8 +34,15 @@ def run(self):
...
+]
+
 EXTRAS_REQUIRE = {
     "datalab": DATALAB_REQUIRE,
+    "cleanvision": CLEANVISION_REQUIRE,
```
Comment (jwmueller) [inline setup.py:45]: I vote to name this key "image" instead. So for installing all optional dependencies related to image data, future cleanlab users can do: `pip install cleanlab[image]`

I think cleanvision is just one special case of this in the long-term.

File: docs/source/tutorials/datalab/datalab_advanced.ipynb [hunk 26]
```diff
@@ -722,7 +722,7 @@
    "outputs": [],
    "source": [
     "from cleanlab.datalab.issue_manager import IssueManager\n",
-    "from cleanlab.datalab.factory import register\n",
+    "from cleanlab.datalab.issue_manager_factory import register\n",
```
Comment (jwmueller) [inline docs/source/tutorials/datalab/datalab_advanced.ipynb:725]: please confirm you've properly updated this import in all of the guides too, eg:
https://docs.cleanlab.ai/stable/cleanlab/datalab/guide/custom_issue_manager.html

CC @elisno

**Review by jwmueller — CHANGES_REQUESTED** [review]
Left some minor suggestions here and there, as well as a couple things to simply confirm. 
Ping me when that's updated and this PR should be good to go.

The biggest thing that is missing is this examples notebook needs to be re-executed after this code is merged, so that its outputs are updated to reflect the ability to run cleanvision issue types inside of datalab:

https://github.com/cleanlab/examples/blob/master/datalab_image_classification/README.md

File: cleanlab/datalab/datalab.py [hunk 27]
```diff
@@ -89,6 +96,7 @@ def __init__(
         self,
         data: "DatasetLike",
         label_name: Optional[str] = None,
+        image_key: Optional[str] = None,
```
Comment (jwmueller) [inline cleanlab/datalab/datalab.py:99]: The documentation for this arg is totally missing...

Here's a suggested docstring:

```
image_key : str
    Optional key that can be specified for image datasets to point to the field containing the actual images themselves.
    If specified, additional image-specific issue types can be detected in the dataset.
    See the CleanVision package documentation for descriptions of these image-specific issue types.
```

File: cleanlab/datalab/datalab.py [hunk 28]
```diff
@@ -21,18 +21,25 @@
 """
 from __future__ import annotations
 
+import warnings
```
Comment (jwmueller) [inline cleanlab/datalab/datalab.py:24]: what happens if user specifies: `image_key` but does not have cleanvision dependencies installed?

File: docs/source/cleanlab/datalab/guide/issue_type_description.rst [hunk 29]
```diff
@@ -107,6 +109,15 @@ The assumption that examples in a dataset are Independent and Identically Distri
...
+Image-specific Issues
+---------------------
+
+For image datasets which are properly specified as such, Datalab can detect additional types of image-specific issues (if the necessary optional dependencies are installed).
+Specifically, low-quality images which are too: dark/bright, blurry, low information, abnormally sized, etc.
+Descriptions of these image-specific issues are provided in the `CleanVision package <https://github.com/cleanlab/cleanvision>`_ and its documentation.
+
+
 Optional Issue Parameters
 =========================
```
Comment (jwmueller) [inline docs/source/cleanlab/datalab/guide/issue_type_description.rst:123]: consider whether we should have an Image-Specific Issue Parameters section in the “Optional Issue Parameters” guide which links to how somebody can specify cleanvision args via datalab.
[https://github.com/sanjanag/cleanlab/blob/cleanvision-integration/docs/source/cleanlab/datalab/guide/issue_type_description.rst#optional[…]arameters](https://github.com/sanjanag/cleanlab/blob/cleanvision-integration/docs/source/cleanlab/datalab/guide/issue_type_description.rst#optional-issue-parameters)

Comment (jwmueller) [reply docs/source/cleanlab/datalab/guide/issue_type_description.rst:123]: cc @elisno

File: .github/workflows/ci.yml (hunk 16 above)
Comment (sanjanag) [reply .github/workflows/ci.yml:145]: Yes, it doesn't affect previous checks, https://github.com/cleanlab/cleanlab/blob/a0d72fcb36f1ce87d1a6022e491e4921c5f4267f/.github/workflows/ci.yml#L103

File: tests/datalab/test_cleanvision_integration.py (hunk 23 above)
Comment (sanjanag) [reply tests/datalab/test_cleanvision_integration.py:1]: This part is already checked in cleanvision repo tests. I didn't add it to avoid duplication.

File: setup.py (hunk 24 above)
Comment (sanjanag) [reply setup.py:40]: I've intentionally kept it fixed for now, since log statements are delicately balanced. This way an updated version will be intentional and require testing.

File: cleanlab/datalab/datalab.py (hunk 28 above)
Comment (sanjanag) [reply cleanlab/datalab/datalab.py:24]: It throws an import error specifying to install image related dependencies or all of them.

File: cleanlab/datalab/datalab.py (hunk 21 above)
Comment (elisno) [reply cleanlab/datalab/datalab.py]: We decided to remove this check+warning as Datalab will just not run anything useful.

As a matter of fact, some issue checks being added don't require and arguments (like the imbalanced classes issue). If it will be a default check, then this warning won't be necessary.

File: tests/datalab/test_cleanvision_integration.py (hunk 23 above)
Comment (jwmueller) [reply tests/datalab/test_cleanvision_integration.py:1]: how can we be sure it's working correctly in datalab?   I don't follow the logic that because its checked in cleanvision repo, there is no way it got messed up in datalab, so I there needs to be at least one end-to-end test for datalab as well

Comment (sanjanag) [reply tests/datalab/test_cleanvision_integration.py:2]: Added an end to end test for datalab as well. Check [here](https://github.com/cleanlab/cleanlab/pull/679/files#diff-810ef4854434604d9f6d7d0b885359d14c915bd387f037dfbcc8b1cc3343acafR38)

File: cleanlab/datalab/adapter/imagelab.py [hunk 30]
```diff
@@ -0,0 +1,198 @@
+"""An internal wrapper around the Imagelab class from the CleanVision package to incorporate it into Datalab.
```
Comment (jwmueller) [inline cleanlab/datalab/adapter/imagelab.py:1]: What is the long-term plan to ensure the top-level datalab documentation is not full files that are only intended for internal use?   CC @elisno 

IMO eventually only datalab.py should be visible at the top-level datalab docs (since this is the only file intended for users to interface with). So everything else could go into a datalab/internal/ folder or datalab/utils/ or something like that so it is buried more deeply in the documentation rather than being visible at the top-level?

Screen Shot 2023-07-16 at 11 14 21 AM

File: tests/datalab/test_cleanvision_integration.py [hunk 31]
```diff
@@ -0,0 +1,200 @@
+import matplotlib.pyplot as plt
+import numpy as np
+import pytest
+import pandas as pd
+
```
Comment (jwmueller) [inline tests/datalab/test_cleanvision_integration.py:5]: wherever you are creating the `image_dataset` out of the files like `tests/datalab/data/random_class_1/image_604.png` etc. Please move those files out of this repo, download them from internet, and then construct the `image_dataset`. This is worthwhile to keep the repo size limited (we hardly ever want to store any data inside this repo)

File: tests/datalab/test_cleanvision_integration.py [hunk 32]
```diff
@@ -0,0 +1,200 @@
...
+        assert (
+            "Finding dark, light, low_information, odd_aspect_ratio, odd_size, grayscale, blurry images"
+            in captured.out
+        )
+        # unable to check for non iid as feature space is too small, skipping it in interest of time
+        assert "Failed to check for these issue types: [NonIIDIssueManager]" in captured.out
+        assert len(datalab.issues) == len(image_dataset)
+        # 14 from imagelab + 6 from datalab
+        assert len(datalab.issues.columns) == 14 + 6
+        assert len(datalab.issue_summary) == 7 + 3
```
Comment (jwmueller) [inline tests/datalab/test_cleanvision_integration.py:51]: where do these magic numbers come from? are they forwards compatible as more issue types are added in datalab? 

At least define them as variables with explanation in a comment, so if the unit test breaks in the future when somebody adds new datalab code, they can more easily update the unit test

Comment (sanjanag) [reply tests/datalab/test_cleanvision_integration.py]: Added variables for these numbers

File: tests/datalab/test_cleanvision_integration.py (hunk 31 above)
Comment (sanjanag) [reply tests/datalab/test_cleanvision_integration.py:5]: moved out to assets repo

File: DEVELOPMENT.md [hunk 33]
```diff
@@ -124,6 +124,14 @@ pip install -e ".[gpu]"
 
 ## Testing
 
+
+**Download test data**
+The test data for cleanlab resides in the assets repository. Use the following commands to download test data before running tests.
+```shell
+git clone https://github.com/cleanlab/assets.git
+mv assets/cleanlab_test_data cleanlab/tests/datalab/data
```
Comment (jwmueller) [inline DEVELOPMENT.md:132]: is this the long-term burden you are imposing on all future developers?

Why is the cleanlab_test_data/ being downloaded into the datalab/data folder?
If this is the long-term plan going forward, shouldn't cleanlab_test_data/ be downloaded into a more universal location that the individual unit-tests then access?

Comment (sanjanag) [reply DEVELOPMENT.md:132]: This does add an extra step for running tests. I downloaded it into datalab, because that's where all the tests for datalab reside. This data is not required for running tests outside of that directory.

Comment (jwmueller) [reply DEVELOPMENT.md:132]: It's ok for this PR but this is not very forwards compatible going forward if other tests want to access cleanlab_test_data, so I recommend just copying cleanlab_test_data into: cleanlab/tests/

Also you can greatly improve developer quality-of-life by fetching this file inside the unit-test that requires it, if it does not already exist locally.

Comment (sanjanag) [reply DEVELOPMENT.md:132]: Okay, I can move it to cleanlab/tests.
I tried fetching it inside the unit test initially, but that was not working quickly for some reason. I'll make a note of it for todos.

**Review by elisno — APPROVED** [review]
Great work @sanjanag!
